kernel/poll.c

```diff
@@ -146,7 +146,14 @@ static inline int register_event(struct k_poll_event *event,
 /* must be called with interrupts locked */
 static inline void clear_event_registration(struct k_poll_event *event)
 {
-	event->poller = NULL;
+        /*
+	 * [Sanechips] If this event has been deleted by clear_event_registrations,
+	 * and there is other thread's event in the event list. Re-delete may cause
+	 * other threads never get scheduled anymore.
+	 */
+	if (event->poller == NULL) {
+	        return ;
+        }
```

Comment:
I don't understand the circumstances where this could happen.  This is an internal function called only out of poll internals, only with interrupts locked, and only when the current function is not polling.  How do you reach the case you're trying to protect against?